### ISSUE TITLE
LM-generated audio codes can exceed codebook range, causing unrecoverable CUDA crash

### ISSUE THREAD

**Issue by chanba** [ace-step/ACE-Step-1.5#92]
### Description

When using the API server for `text2music` generation, the LM occasionally generates audio code tokens with indices outside the valid range of the `ResidualFSQ` codebook (0–63999). This triggers a CUDA assertion in `IndexKernel.cu` that **poisons the entire CUDA context**, causing all subsequent jobs to fail until the process is fully restarted.

The probability increases with higher `lm_cfg_scale` values, as classifier-free guidance pushes logits harder during decoding.

### Environment

- GPU: NVIDIA RTX PRO 6000 Blackwell Max-Q (95GB VRAM)
- DiT model: `acestep-v15-sft` (50 steps, shift 3.0)
- LM model: `acestep-5Hz-lm-4B` (also reproduced with 0.6B)
- Task: `text2music` with English lyrics, ~3.5 min duration

### Steps to Reproduce

1. Start the API server with SFT model
2. Submit `text2music` jobs with `lm_cfg_scale >= 3.0`
3. Some jobs generate codes with out-of-range indices (e.g., 65506 vs max 63999)
4. CUDA assertion fires → CUDA context poisoned → all subsequent jobs fail
5. Only fix is full process restart

### Error

```
/pytorch/aten/src/ATen/native/cuda/IndexKernel.cu:111:
  Assertion `-sizes[i] <= index && index < sizes[i] && "index out of bounds"` failed.
```

Traceback:
```
File "acestep/handler.py", line 761, in _decode_audio_codes_to_latents
    quantized = quantizer.get_output_from_indices(indices)
```

### Failure rate vs lm_cfg_scale

Tested across multiple genres (indie, jazz, hip hop, country, orchestral, bossa nova, synthwave):

| lm_cfg_scale | Jobs | Failures | Rate |
|:-:|:-:|:-:|:-:|
| 2.0 | 5 | 0 | 0% |
| 2.5 | 4 | 0 | 0% |
| 3.0 | 6 | 3 | ~50% (without fix) |

### Proposed Fix

Add a bounds check + clamp in `_decode_audio_codes_to_latents()` after creating the indices tensor and before passing to the quantizer. **Clamp** out-of-range indices to the valid range rather than raising an error — only a few codes out of thousands are affected per generation, and clamping produces usable audio with minimal quality impact.

```python
# In handler.py, _decode_audio_codes_to_latents method
# After: indices = torch.tensor(code_ids, device=self.device, dtype=torch.long)
# Before: quantized = quantizer.get_output_from_indices(indices)

codebook_size = getattr(quantizer, 'codebook_size', None)
if codebook_size is None:
    levels = getattr(quantizer, 'levels', None)
    if levels is not None:
        codebook_size = 1
        for level in levels:
            codebook_size *= level

if codebook_size is not None:
    max_index = codebook_size - 1
    out_of_range = (indices > max_index) | (indices < 0)
    if out_of_range.any():
        logger.warning(
            "Audio code indices out of bounds (clamping %d/%d codes): "
            "min=%d, max=%d, valid_range=[0, %d]. "
            "High lm_cfg_scale can cause this; consider reducing it.",
            out_of_range.sum().item(), indices.numel(), max_index,
            indices.min().item(), indices.max().item()
        )
        indices = indices.clamp(0, max_index)
```

### Why clamp instead of error?

We initially shipped a `raise ValueError` version that prevented CUDA crashes but still failed the generation. After switching to clamping:

- **6/6 stress test at lm_cfg_scale 3.0 and 3.5** — all succeeded
- Audio quality is indistinguishable from clean generations (only a few codes out of thousands are affected)
- Server stays alive, no CUDA context poisoning
- The clamped codes map to nearby valid codebook entries, producing minimal artifacts

### Impact without fix

- CUDA assertion poisons the entire CUDA context for the process
- Every subsequent job fails regardless of input
- Only recovery is full process restart
- Makes the API server unreliable for any batch/sequential workload
- Higher `lm_cfg_scale` values (which produce more creative/complex arrangements) become unusable
